src/VisualStudio/Core/Def/Implementation/Venus/ContainedLanguage.cs

```diff
@@ -146,7 +146,11 @@ internal partial class ContainedLanguage<TPackage, TLanguageService>
 
             // Create our tagger
             var bufferTagAggregatorFactory = ComponentModel.GetService<IBufferTagAggregatorFactoryService>();
-            _bufferTagAggregator = bufferTagAggregatorFactory.CreateTagAggregator<ITag>(SubjectBuffer);
+
+            // Make sure we only get taggers on the given buffer not all the projection buffers involved to
+            // compose the subject buffer
+            _bufferTagAggregator = bufferTagAggregatorFactory.CreateTagAggregator<ITag>(
+                SubjectBuffer, (TagAggregatorOptions)TagAggregatorOptions2.NoProjection);
```

Comment:
❔ Is this required now that we are disposing of the aggregator in `OnDisconnect`? It seems like any unnecessary taggers held by the aggregator would be held by other sources at least until after `OnDisconnect` is called, in which case this change might not buy us anything.

If this change *is* required for correctness, we should investigate why a public API (buffer tag aggregator) has a dependency on a non-public API for correctness.

Reply:
we created the aggregator so we own the aggregator. so not disposing it is a bug regardless of any other thing.

I have no idea what you meant by "we should investigate why a public API (buffer tag aggregator) has a dependency on a non-public API for correctness."

Reply:
@sharwell's point is "why are we using Microsoft.VisualStudio.Text.Internal".

In any case, there are two changes here. This is just an optimization -- the only point of this is to ensure our taggers cache states for our own buffer. Any taggers on buffers lower than us we don't care about. But it's not needed for correctness.

Reply:
(Put another way: the correctness doesn't depend on an internal API. Another suggested improvement does.)

Reply:
I see. when they want to make the option2 public will be editor team's choice. tagging @dpugh